### PR TITLE
agent: simplify codes

### PR DESCRIPTION
**Pull request by liubin**
Fixes: #922

### CODE REVIEW

**Review by jodh-intel — APPROVED** [review]
Thanks @liubin !

**Review by Tim-Zhang — APPROVED** [review]

File: src/agent/src/rpc.rs [hunk 1]
```diff
@@ -251,15 +248,14 @@ impl agentService {
...
-                None => {
-                    return;
-                }
-            };
-
-            ctr.destroy().unwrap();
-            tx.send(1).unwrap();
+            let _ctr = sandbox
+                .get_container(&cid2)
+                .ok_or(anyhow!("Invalid container id"))
```
Comment (Tim-Zhang) [inline src/agent/src/rpc.rs:253]: better to use `?` to replace and_then

Comment (liubin) [reply src/agent/src/rpc.rs:253]: It's in a closure, so can't use `?`

**Review by bergwolf — APPROVED** [review]

**Pull request merged**